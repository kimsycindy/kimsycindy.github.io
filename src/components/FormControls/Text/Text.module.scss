@import '../../../styles/variables.scss';

.input {
  background-color: $gray-light;
  padding: 10px;
  width: 100%;
}
