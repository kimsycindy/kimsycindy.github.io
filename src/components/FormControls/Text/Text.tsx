import React from 'react'
import styles from './Text.module.scss'

interface TextProps {
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  type: 'email' | 'text'
  name: string
  label: string
  value: string
  placeholder?: string
  required?: boolean
}

export const Text = ({
  onChange,
  type,
  name,
  label,
  value,
  placeholder,
  required = false,
}: TextProps) => (
  <div>
    <label htmlFor={name}>
      {label}
      <br />
      <input
        className={styles.input}
        onChange={onChange}
        type={type}
        name={name}
        value={value}
        placeholder={placeholder}
        required={required}
      />
    </label>
  </div>
)

export type FormControlsTextElement = HTMLInputElement
