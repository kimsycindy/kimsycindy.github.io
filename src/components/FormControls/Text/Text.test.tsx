import React from 'react'
import renderer from 'react-test-renderer'

import { Text } from './Text'

describe('FormControls Text', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Text
          onChange={jest.fn()}
          type="text"
          name="mock"
          label="Mock"
          value=""
          placeholder="Placeholder"
          required
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
