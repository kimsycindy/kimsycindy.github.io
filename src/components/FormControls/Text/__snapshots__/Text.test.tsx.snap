// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControls Text renders correctly 1`] = `
<div>
  <label
    htmlFor="mock"
  >
    Mock
    <br />
    <input
      className="input"
      name="mock"
      onChange={[MockFunction]}
      placeholder="Placeholder"
      required={true}
      type="text"
      value=""
    />
  </label>
</div>
`;
